gpustack/policies/candidate_selectors/vllm_resource_fit_selector.py

```diff
@@ -377,6 +419,36 @@ async def _find_single_worker_multi_gpu_full_offloading_candidates(  # noqa: C90
             reverse=True,
         )
 
+        if self._selected_gpu_worker:
+            if worker.name != self._selected_gpu_worker:
+                return []
+
+            if (
+                self._num_attention_heads
+                and self._num_attention_heads % len(self._selected_gpu_indexes) != 0
+            ):
+                return []
```

Comment:
What's the UX for this case, can user tell why scheduling fails?

Reply:
Update the error message to instance state message